marshal/claim.go

```diff
@@ -229,32 +239,18 @@ func (c *claim) GetCurrentLag() int64 {
 	return 0
 }
 
-// Release will invoke our release mechanism if and only if we are still claimed.
-func (c *claim) Release() bool {
-	if !atomic.CompareAndSwapInt32(c.claimed, 1, 0) {
+// Release will invoke commit offsets and may release the Kafka partition depending on whether
+// releasePartition was passed or not
+func (c *claim) Release(releasePartition bool) bool {
```

Comment:
Sounds good...I just wanted to share the implementation...I will create two separate functions...